hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -380,9 +380,10 @@ def action_timesheet_draft(self):
                 _('Only an HR Officer or Manager can refuse sheets '
                   'or reset them to draft.'))
         self.write({'state': 'draft'})
-        domain = self._get_timesheet_sheet_lines_domain()
-        timesheets = self.env['account.analytic.line'].search(domain)
-        self.clean_timesheets(timesheets)
+        for sheet in self:
+            domain = sheet._get_timesheet_sheet_lines_domain()
+            timesheets = self.env['account.analytic.line'].search(domain)
+            self.clean_timesheets(timesheets)
```

Comment:
I think that https://github.com/OCA/hr-timesheet/commit/640d9289d5f9ece39cb93e2bc144b8cb892ffdde#diff-a48ce9b933da452cb1910c0e256932c7 makes the `clean_timesheets()` not necessary anymore here.


`clean_timesheets()` does two things:

1. sets the proper `sheet_id` in the lines in case it's wrong or not set;
2. merges the lines with description = `/`


Point 1 should be non needed anymore after https://github.com/OCA/hr-timesheet/commit/640d9289d5f9ece39cb93e2bc144b8cb892ffdde#diff-a48ce9b933da452cb1910c0e256932c7 because the computation of `sheet_id` is now in the create and the write of the lines. If I'm not wrong, this part of code is now obsolete and can be safely removed.

Point 2 could be also removed IMO when resetting to draft, as proposed in #197

Reply:
Yeah, removed the `clean_timesheets` in the "set to draft" method

Reply:
Thanks!